src/main/kotlin/com/hibob/academy/resource/petsResource.kt

```diff
@@ -0,0 +1,48 @@
+package com.hibob.academy.resource
+
+import com.hibob.academy.types.Owner
+import com.hibob.academy.types.Pets
+import jakarta.ws.rs.*
+import jakarta.ws.rs.core.Response
+import org.springframework.stereotype.Controller
+import jakarta.ws.rs.core.MediaType
+import org.springframework.web.bind.annotation.RequestBody
+
+@Controller
+@Path("api/pets")
+@Produces(MediaType.APPLICATION_JSON)
+class PetsResource {
+
+    @POST
+    @Path("/type")
+    @Consumes(MediaType.APPLICATION_JSON)
```

Comment:
What does type do here as well? You're not creating a new type here. You're creating a new pet. It's confusing.